tfx/utils/path_utils.py

```diff
@@ -23,60 +23,36 @@
 
 import tensorflow as tf
 
-from tfx.utils import io_utils
-
 EVAL_MODEL_DIR = 'eval_model_dir'
 SERVING_MODEL_DIR = 'serving_model_dir'
 
-# TODO(b/127149760): simplify this PPP-esque structure.
-#
-# Directory structure of exported model for estimator based trainer:
-#   |-- <ModelExportPath>
-#       |-- EVAL_MODEL_DIR  <- eval_model_dir
-#           |-- <timestamped model>  <- eval_model_path
-#               |-- saved_model.pb
-#               |-- ...
-#       |-- SERVING_MODEL_DIR  <- serving_model_dir
-#           |-- export
-#               |-- <exporter name>
-#                   |-- <timestamped model>  <- serving_model_path
-#                       |-- saved_model.pb
-#                       |-- ...
-#           |-- ...
-#
-# For generic trainer with Keras, there won't be eval model:
-#   |-- <ModelExportPath>
-#       |-- SERVING_MODEL_DIR  <- serving_model_dir
-#           |-- saved_model.pb
-#           |-- ...
-
-
-def eval_model_dir(output_uri: Text) -> Text:
-  """Returns directory for exported model for evaluation purpose."""
-  return os.path.join(output_uri, EVAL_MODEL_DIR)
-
+"""
+Directory structure of exported model for estimator based trainer:
+  |-- <ModelExportPath>
+      |-- EVAL_MODEL_DIR  <- eval_model_dir, eval_model_path
+          |-- saved_model.pb
+          |-- ...
+      |-- SERVING_MODEL_DIR  <- serving_model_dir, serving_model_path
+          |-- saved_model.pb
+          |-- ...
+
+For generic trainer with Keras, there won't be eval model:
+  |-- <ModelExportPath>
+      |-- SERVING_MODEL_DIR  <- serving_model_dir, serving_model_path
+          |-- saved_model.pb
+          |-- ...
+"""
 
 def eval_model_path(output_uri: Text) -> Text:
-  """Returns path to timestamped exported model for evaluation purpose."""
-  model_dir = eval_model_dir(output_uri)
+  """Returns path to exported model for evaluation purpose."""
+  model_dir = os.path.join(output_uri, EVAL_MODEL_DIR)
   if tf.io.gfile.exists(model_dir):
-    return io_utils.get_only_uri_in_dir(model_dir)
+    return model_dir
   else:
     # If eval model doesn't exist, use serving model for eval.
-    return serving_model_dir(output_uri)
-
-
-def serving_model_dir(output_uri: Text) -> Text:
-  """Returns directory for exported model for serving purpose."""
-  return os.path.join(output_uri, SERVING_MODEL_DIR)
+    return serving_model_path(output_uri)
 
 
 def serving_model_path(output_uri: Text) -> Text:
```

Comment:
ideally we want the new version but we still need this for back compat I think

we want sth like
```
serving_model_path():
if old structure:
  log warning, mention we are going to deprecate the support of this structure, please have xxx structure
  return model path
if new structure:
  return model path
```